client/question/question.js

```diff
@@ -20,6 +20,71 @@ window.tie = angular.module('tie',
   ['ui.codemirror', 'tieConfig', 'tieData', 'ngCookies']);
 
 /**
+ * String used to label test suites that contain sample input and output
+ * that are presented to the user.
+ *
+ * @type {string}
+ */
+tie.constant('SAMPLE_INPUT', 'SAMPLE_INPUT'); 
+
+/**
+ * Correctness state where the test case input has been displayed to the user.
+ *
+ * @type {string}
+ */
+tie.constant('CORRECTNESS_TEST_INPUT_DISPLAYED',
```

Comment:
Ditto; start with the type of what this is. Maybe CORRECTNESS_STATE_KEY_*.

Reply:
I thought I was following the naming convention you suggested for string constants denoting the stages of feedback? I used 'DISPLAYED' instead of 'GIVEN'.

Reply:
Ah yep, this one for example should be CORRECTNESS_STATE_KEY_INPUT_DISPLAYED (that's what I meant by the previous notation).

Reply:
Just to make sure we talking about the same thing, you suggested in a prior comment to use:
"CORRECTNESS_TEST_INPUT_GIVEN, CORRECTNESS_TEST_EXPECTED_OUTPUT_GIVEN, CORRECTNESS_TEST_OBSERVED_OUTPUT_GIVEN". I changed GIVEN to DISPLAYED but used "CORRECTNESS_TEST_*". Are now you saying use "CORRECTNESS_STATE_KEY_*" instead or is this something different? If we are talking about the same thing, would "CORRECTNESS_STATE_*" work or do you think we really should have "_KEY" as well? Though I don't have strong feelings about it.

Reply:
Not sure why "Are now you saying use" is italicized; not trying to emphasize it. Can't seem to change it back to normal text.

Reply:
It's a markdown thing; enclosing anything between asterisks italicizes it. A fix would be to use backticks (`) to surround parts that should be displayed in code form.

I also got the name wrong earlier, sorry. I think CORRECTNESS_STATE or CORRECTNESS_STATE_KEY would be fine. Happy with the former.

Reply:
Changed to 'CORRECTNESS_STATE_*'